test/integration/scheduler_perf/util.go

```diff
@@ -17,33 +17,50 @@ limitations under the License.
 package benchmark
 
 import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	clientset "k8s.io/client-go/kubernetes"
 	restclient "k8s.io/client-go/rest"
-	"k8s.io/kubernetes/pkg/scheduler/factory"
 	"k8s.io/kubernetes/test/integration/util"
 )
 
 // mustSetupScheduler starts the following components:
 // - k8s api server (a.k.a. master)
 // - scheduler
-// It returns scheduler config factory and destroyFunc which should be used to
+// It returns clientset and destroyFunc which should be used to
 // remove resources after finished.
 // Notes on rate limiter:
 //   - client rate limit is set to 5000.
-func mustSetupScheduler() (*factory.ConfigFactoryArgs, util.ShutdownFunc) {
+func mustSetupScheduler() (util.ShutdownFunc, clientset.Interface) {
 	apiURL, apiShutdown := util.StartApiserver()
 	clientSet := clientset.NewForConfigOrDie(&restclient.Config{
 		Host:          apiURL,
 		ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Group: "", Version: "v1"}},
 		QPS:           5000.0,
 		Burst:         5000,
 	})
-	schedulerConfigArgs, schedulerShutdown := util.StartScheduler(clientSet)
+	_, schedulerShutdown := util.StartScheduler(clientSet)
 
 	shutdownFunc := func() {
 		schedulerShutdown()
 		apiShutdown()
 	}
-	return schedulerConfigArgs, shutdownFunc
+	return shutdownFunc, clientSet
+}
+
+func getScheduledPods(clientset clientset.Interface) ([]*v1.Pod, error) {
+	podList, err := clientset.CoreV1().Pods("").List(metav1.ListOptions{})
```

Comment:
`getScheduledPods()` is on the critical path. We shouldn't call apiserver heavily.

Reply:
done